compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -30,6 +30,13 @@ class HAL_MakeTupleOp<string mnemonic, list<Trait> traits = []> :
 //===----------------------------------------------------------------------===//
 // TODO(benvanik): remove these as the sequencer/other HAL ops are added.
 
+def OpGroupExperimentalOps : OpDocGroup {
+  let summary = "Experimental ops";
+  let description = "Temporary hack ops expected to be removed in the future.";
+}
+
+let opDocGroup = OpGroupExperimentalOps in {
+
 def HAL_ExSharedDeviceOp : HAL_PureOp<"ex.shared_device", [
```

Comment:
@jpienaar this group only has one op, so the `bool nested = grouping.ops.size() > 1;` check in https://reviews.llvm.org/D153306 doesn't group it:

Here's what we see:  ![image](https://github.com/openxla/iree/assets/4010439/9fdc6846-78bb-481e-8830-b3035b59faa9)

Here's what I'd expect: ![image](https://github.com/openxla/iree/assets/4010439/15c4f9cf-7351-42f0-89f7-778e708bf6a4)

Reply:
Funnily I had a change for that and then removed it. I'll update.